iam.yml

```diff
@@ -10,13 +10,137 @@ Parameters:
   TemplateBucket:
     Type: String
     Description: Name of existing S3 bucket for storing CloudFormation template artifacts
+  JavabuilderApiId:
+    Type: String
+    Description: The API ID of the Javabuilder API Gateway V2 Api
+    Default: "*"
 
 Resources:
-  # TODO: Split this into separate roles:
-  # 1) Permissions for the long-running Lambda that compiles/runs student projects
-  # 2) Permissions for the synchronous Lambda that invokes the long-running Lambda and then relays web messages to it.
-  # 3) Permissions for the Lambda Authorizer
-  # 4) Permissions for CodePipeline Slack approval Lambda.
+  BuildAndRunLambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      Path: /
+      AssumeRolePolicyDocument:
+        Statement:
+          - Action: ['sts:AssumeRole']
+            Effect: Allow
+            Principal: {Service: [lambda.amazonaws.com]}
+      Policies:
+        - PolicyName: BuildAndRunLambdaExecutionPolicy
+          PolicyDocument:
+            Statement:
+              # BuildAndRunJavaProject Lambda needs to put objects to the content bucket.
+              - Effect: Allow
+                Action:
+                  - 's3:PutObject'
```

Comment:
Do we need additional permission to get signed URL's too?

Reply:
Yes, but I don't see where we gave permission to do that in the past. (sorry, not the most helpful response)

Reply:
At the moment, I don't think Javabuilder is actually GET-ing any signed URLs? We read and write to the output bucket and generate signed urls for the client, but I don't think we ever read signed URLs for another bucket?

Reply:
for some reason I was thinking that you needed an explicit getPresignedUrl permission, but that's not a thing, it just uses your current Get/Put permissions.